azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Route.java

```diff
@@ -0,0 +1,243 @@
+         */
+        interface WithNextHopToVirtualAppliance<ParentT> {
+            /**
+             * Specifies the IP address of the virtual appliance for the next hop to go to.
+             * @param ipAddress an IP address of an existing virtual appliance (virtual machine)
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withNextHopToVirtualAppliance(String ipAddress);
+        }
+
+        /**
+         * The stage of a route definition allowing to specify the next hop type.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithNextHopType<ParentT> extends WithNextHopToVirtualAppliance<ParentT> {
+            /**
+             * Specifies the next hop type.
+             * <p>
+             * To use a virtual appliance, use {@link WithNextHopToVirtualAppliance#withNextHopToVirtualAppliance()} instead and specify its IP address.
+             * @param nextHopType a hop type
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withNextHop(RouteNextHopType nextHopType);
+        }
+    }
+
+    /** The entirety of a route definition.
+     * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+     */
+    interface Definition<ParentT> extends
+        DefinitionStages.Blank<ParentT>,
+        DefinitionStages.WithAttach<ParentT>,
+        DefinitionStages.WithNextHopType<ParentT>,
+        DefinitionStages.WithDestinationAddressPrefix<ParentT>,
+        DefinitionStages.WithNextHopToVirtualAppliance<ParentT> {
+    }
+
+    /**
+     * Grouping of route update stages.
+     */
+    interface UpdateStages {
+        /**
+         * The stage of a route update allowing to modify the destination address prefix.
+         */
+        interface WithDestinationAddressPrefix {
+            /**
+             * Specifies the destination address prefix to apply the route to.
+             * @param cidr an address prefix expressed in the CIDR notation
+             * @return the next stage of the update
+             */
+            Update withDestinationAddressPrefix(String cidr);
+        }
+
+        /**
+         * The stage of a route update allowing to modify the IP address of a virtual appliance to
+         * direct the traffic through.
+         */
+        interface WithNextHopToVirtualAppliance {
+            /**
+             * Specifies the IP address of the virtual appliance for the next hop to go to.
+             * @param ipAddress an IP address of an existing virtual appliance (virtual machine)
+             * @return the next stage of the update
+             */
+            Update withNextHopToVirtualAppliance(String ipAddress);
+        }
+
+        /**
+         * The stage of a route update allowing to specify the next hop type.
+         */
+        interface WithNextHopType extends WithNextHopToVirtualAppliance {
```

Comment:
Why we are extending 'WithNextHopType' from 'WithNextHopToVirtualAppliance' in case of Update? the Update interface already extends from these two interfaces explicitly ?